### PR TITLE
Refactor: test helper schema

### PR DESCRIPTION
**Pull request by dleard**
There is documentation in `docs/test-helper.md` on the test-helper schema and functions.

There are a lot of changed files in this PR, but most of the work that needs review is under the `test_helper_schema` directory. The changes in the `fixtures/` and `cypress/` directories are almost all deletions/refactoring the fixtures to use the new schema.

All Cypress specs & their fixtures should now run independently. All teardown files have been removed.

### CODE REVIEW

**Comment by dleard** [comment]
@kriscooke tagging you in this draft PR

**Review by matthieu-foucault — APPROVED** [review]
Looks great!
I'd like to find where those extra 8min of CI time come from before merging this.

**Comment by matthieu-foucault** [comment]
@pbastia I see that you have a linter issue. Is your pre-commit hook working properly?

File: app/cypress/integration/reporter-access-all-pages.spec.js [hunk 1]
```diff
@@ -40,6 +41,7 @@ describe('When logged in as a reporter', () => {
         cy.url().should('include', '/certifier/certification-redirect');
         cy.get('#page-content');
         cy.contains('Continue').click();
+        cy.wait(500);
```
[the remark this answers is not in the document]
Comment (dleard) [reply app/cypress/integration/reporter-access-all-pages.spec.js:44]: I can go through and look at the `cy.wait()` calls to see if they are needed. I think many of them are fixing a local issue of some test stuff running before it should. May not be needed for CI.

[the remark this answers is not in the document]
Comment (dleard) [reply app/cypress/integration/reporter-access-all-pages.spec.js:44]: I've added a new tech debt card to audit the use of cy.wait() https://youtrack.button.is/issue/GGIRCS-2039

**Comment by matthieu-foucault** [comment]
@pbastia @dleard Let's get this merged and look at e2e test performance soon. This needs a rebase, and I see that the linter is failing as well

**Review by matthieu-foucault — APPROVED** [review]

**Pull request merged**